debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -198,14 +198,29 @@ protected void recoverRecords(Tables schema, DdlParser ddlParser, Consumer<Histo
                         Long partition = new Long(record.partition());
                         Long lastOffset = offsetsByPartition.get(partition);
                         if (lastOffset == null || lastOffset.longValue() < record.offset()) {
-                            HistoryRecord recordObj = new HistoryRecord(reader.read(record.value()));
-                            records.accept(recordObj);
-                            logger.trace("Recovered database history: {}" + recordObj);
+                            if (record.value() == null) {
+                                logger.warn("Skipping null database history record. " +
+                                        "This is often not an issue, but if it happens repeatedly please check the '{}' topic.");
+                            } else {
+                                HistoryRecord recordObj = new HistoryRecord(reader.read(record.value()));
+                                logger.trace("Recovering database history: {}" + recordObj);
+                                if (recordObj == null || !recordObj.isValid()) {
+                                    logger.warn("Skipping invalid database history record '{}'. " +
+                                            "This is often not an issue, but if it happens repeatedly please check the '{}' topic.",
+                                            recordObj, topicName);
+                                } else {
+                                    records.accept(recordObj);
+                                    logger.trace("Recovered database history: {}" + recordObj);
```

Comment:
The log calls should pass the parameters as arguments, not using String concatenation. Looks like it was wrong in the original code.